dali/operators/input/video_input.h

```diff
@@ -67,6 +69,55 @@ auto DetermineBatchOutline(int num_frames, int frames_per_sequence, int batch_si
   return std::make_tuple(num_full_batches, num_full_sequences, frames_in_last_sequence);
 }
 
+
+/**
+ * Helper structure to handle Pad frames.
+ *
+ * Pad frame is a frame, which will be used for padding incomplete sequences.
+ *
+ * This structure, given the value and shape, generates a pad frame on a proper device.
+ *
+ * @tparam OutBackend Backend, on which the pad frame will be allocated.
+ * @tparam PadType Type of the pad value.
+ */
+template<typename OutBackend, typename PadType>
+struct PadFrameCreator {
+  static constexpr bool is_cpu = std::is_same_v<OutBackend, CPUBackend>;
+
+  template<typename T>
+  using container_type =
+          std::conditional_t<is_cpu, thrust::host_vector<T>, thrust::device_vector<T>>;
+
+  PadFrameCreator() = default;
+
+  PadFrameCreator(PadType value, TensorShape<3> frame_shape) {
+    Initialize(value, frame_shape);
+  }
+
+
+  void Initialize(PadType value, TensorShape<3> frame_shape) {
+    pad_frame_shape_ = frame_shape;
+    if constexpr (is_cpu) {
+      pad_frame_data_ = thrust::host_vector<PadType>{
+              static_cast<size_t>(pad_frame_shape_.num_elements()), value};
+    } else {
+      thrust::host_vector<PadType> tmp{static_cast<size_t>(pad_frame_shape_.num_elements()), value};
+      pad_frame_data_ = tmp;
```

Comment:
How is that synchronized? How it is sync with the workspace's stream?

Reply:
Changed to Tensor, that handles synchronization